### PR TITLE
some of the ref access_keys were generating as the same

### PR DESCRIPTION
**Pull request by andrewroth**
@twinge 

made it a lot more aggressive to not have any duplicates.  Seem
important otherwise people could just change their id by one and with
the same access_key they can access someone else's ref

I added these to the string the key is built off:

1 - milliseconds to the generate time
2 - the largest id in the reference sheets table
3 - a nonce on at the class level (in case generate_access_key happens
to get called multiple times before a save happens that would
incremement the largest id in the table)

### CODE REVIEW

File: app/models/fe/reference_sheet.rb [hunk 1]
```diff
@@ -76,7 +76,8 @@ class ReferenceSheet < ActiveRecord::Base
 
     alias_method :applicant, :applicant_answer_sheet
     def generate_access_key
-      self.access_key = Digest::MD5.hexdigest(email + Time.now.to_s)
+      @@nonce ||= 0
+      self.access_key = Digest::MD5.hexdigest(email + Time.now.strftime("%G-W%V-%uT%T%:z%L") + Fe::ReferenceSheet.maximum(:id).to_s + (@@nonce += 1).to_s)
```
Comment (twinge) [inline app/models/fe/reference_sheet.rb]: I prefer this pattern: https://github.com/CruGlobal/sp/blob/394c1ddf458a7ff25bdca361cb7810017f96be84/app/models/api_key.rb#L6

What you did is definitely a big improvement, but that pattern will be 100%

Comment (andrewroth) [inline app/models/fe/reference_sheet.rb]: Yeah I like that better too. Simpler. I'll update this later with it.

**Comment by andrewroth** [comment]
@twinge how's this?

**Comment by twinge** [comment]
why make it a module? are you going to use it in multiple places?

**Comment by andrewroth** [comment]
Yeah, I said in the commit.. I'm using it in some enclosing apps,
specifically the cap uses an access key for the pre-interview form

**Comment by twinge** [comment]
Sorry for missing that. :+1:

**Pull request closed without merging**